test/com/google/enterprise/adaptor/opentext/OpentextAdaptorTest.java

```diff
@@ -2951,6 +2954,116 @@ public void testGetPublicAccessGroupDisabledUser()
+    assertEquals(Lists.newArrayList(
+            newUserPrincipal("user0"), newUserPrincipal("user1"),
+            newUserPrincipal("user2"), newUserPrincipal("user3"),
+            newUserPrincipal("user4")),
+        groupDefinitions.get(newGroupPrincipal("group0")));
+    assertEquals(Lists.newArrayList(
+            newUserPrincipal("user5"), newUserPrincipal("user6"),
+            newUserPrincipal("user7"), newUserPrincipal("user8"),
+            newUserPrincipal("user9")),
+        groupDefinitions.get(newGroupPrincipal("group1")));
+    assertEquals(Lists.newArrayList(
+            newUserPrincipal("user10"), newUserPrincipal("user11"),
+            newUserPrincipal("user12"), newUserPrincipal("user13"),
+            newUserPrincipal("user14")),
+        groupDefinitions.get(newGroupPrincipal("group2")));
+    assertEquals(Lists.newArrayList(
+            newUserPrincipal("user15"), newUserPrincipal("user16"),
+            newUserPrincipal("user17"), newUserPrincipal("user18"),
+            newUserPrincipal("user19")),
+        groupDefinitions.get(newGroupPrincipal("group3")));
+
+    soapFactory.memberServiceMock.deleteMember(2002);
+    pusher = new RecordingDocIdPusher();
+    adaptor.getDocIds(pusher);
+    groupDefinitions = pusher.getGroupDefinitions();
+    assertEquals(null, groupDefinitions.get(newGroupPrincipal("group2")));
+  }
+
+  @Test
+  public void testGroupsError() throws InterruptedException {
+    class MemberServiceMockError extends MemberServiceMock {
+      boolean called;
+      public MemberSearchResults getSearchResults(PageHandle pageHandle) {
+        if (!called)  {
+          called = true;
+          return new MemberSearchResultsMock((PageHandleMock) pageHandle);
+        }
+        throw getSoapFaultException("MemberService SOAPFaultException",
+            "urn:opentextadaptortest", "Test.MemberServiceException", "ns0");
+      }
+    }
+    SoapFactoryMock soapFactory = new SoapFactoryMock();
+    soapFactory.memberServiceMock = new MemberServiceMockError();
+    for (int i = 0; i < 20; i++) {
+      soapFactory.memberServiceMock.addMember(
+          getMember(1000 + i, "user" + i, "User"));
+    }
+    // The search uses a page size of 50, and we want to trigger
+    // an exception on the second page in order to get partial
+    // results, so create a bunch of groups.
+    for (int i = 0; i < 64; i++) {
+      soapFactory.memberServiceMock.addMember(
+          getMember(2000 + i, "group" + i, "Group"));
+      soapFactory.memberServiceMock.addMemberToGroup(
+          2000 + i,
+          soapFactory.memberServiceMock.getMemberById(1000 + (i % 20)));
+    }
+    OpentextAdaptor adaptor = new OpentextAdaptor(soapFactory);
+    AdaptorContext context = ProxyAdaptorContext.getInstance();
+    Config config = initConfig(adaptor, context);
+    adaptor.init(context);
+
+    class MyRecordingDocIdPusher extends RecordingDocIdPusher {
+      DocIdPusher.FeedType feedType;
```

Comment:
Maybe s/feedType/latestFeedType/? We had intentionally designed RecordingDocIdPusher to make assertions about the state, not the behavior (e.g., by pushing disjoint sets of groups, so that we could see that the second one must not have been full, rather than making assertions about the parameters to pushGroupDefinitions), but I think this is OK.